source/opt/strength_reduction_pass.cpp

```diff
@@ -110,7 +110,9 @@ bool StrengthReductionPass::ReplaceMultiplyByPowerOf2(
         context()->ReplaceAllUsesWith(inst->result_id(), newResultId);
 
         // Remove the old instruction.
-        context()->KillInst(&*inst);
+        ir::Instruction* inst_to_delete = &*inst;
+        --inst;
```

Comment:
Unnecessary, "inst" is not used again.

Reply:
|inst| is a reference to the iterator that is pointer to by |instPtr|.  It is needed.  The code is confusing.  I'll fix it up.